proto/gds/peers/v1/peers.proto

```diff
@@ -0,0 +1,34 @@
+syntax = "proto3";
+
+package gds.peers.v1;
+option go_package = "github.com/trisacrypto/directory/pkg/gds/peers/v1;peers";
+
+// Peer contains metadata about how to connect to remote peers in the directory service
+// network. This message services as a data-transfer and exchange mechanism for dynamic
+// networks with changing membership.
+message Peer {
+    uint64 id = 1;                   // the process id of the peer must be unique in the network; used for distributed versions
+    string addr = 2;                 // the network address to connect to the peer on (don't forget the port!)
+    string name = 3;                 // optional - a unique, human readable name for the peer
+    string region = 4;               // optional - the region the peer is running in
+
+    // extra information that might be relevant to process-specific functions; e.g. for
+    // specific clouds or data that's been parsed (optional).
+    map<string, string> extra = 15;
+}
```

Comment:
Pretty basic Peer definition. I've added an extra field for storing arbitrary information, e.g. maybe cloud-specific or k8s information. I thought about storing anti-entropy information here as well; but I think all of that would be tracked in memory. Thoughts?

Reply:
Perhaps we should make a backlog issue to look into what we can get from kubernetes so we have a better notion of what might populate this map; I know we had a task like that in the backlog for another project but didn't get around to it yet,

Reply:
Issue made,  #55